go/arrow/flight/doc.go

```diff
@@ -0,0 +1,98 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Here we list best practices and common pitfalls for Arrow Flight usage.
+//
+// GRPC
```

Comment:
Typically the package comment should begin with `Package <name>` so we should first start with:

`Package flight contains server and client implementations for the Arrow Flight RPC` and any other description of the package. Then follow to the best practices and pitfalls after that.

The reason for this is that the first sentence will show up in the documentation as a summary for the package:

![image](https://user-images.githubusercontent.com/555095/185425569-f268f22a-f66c-4463-9737-72ccf0f3da03.png)

Reply:
Got it! That's nice to know. Writing go docs for the first time is interesting.